content/blog/lift-and-transform-apps-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,283 @@
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  description: Volume transformed from vVol by HPE CSI Driver
+  allowOverrides: importVolAsClone,importVolumeName,forceImport
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+The important details in the `StorageClass` are the `allowOverrides`. Here we put the parameters we want the application admin to override once testing has started (`importVolAsClone`) and once the database is ready for the final import (`importVolumeName` and `forceImport`).
+
+For the final production instance, we’re going to use the [Bitnami MariaDB Helm chart](https://artifacthub.io/packages/helm/bitnami/mariadb). For this chart to install cleanly and be able to bring up the database, the following considerations needs to be taken into account:
+
+* There needs to be a known `root@localhost` account with a known password in the source database.
+* The Bitnami chart does not support altering the `mountPath`. The source filesystem needs to have a symlink to `/var/lib/mysql` from `/var/lib/data`.
+
+For the initial cloning of the source database, the following `values.yaml` file is being used.
+
+```yaml
+image:
+  tag: 10.3
+auth:
+  rootPassword: my-password
+global:
+  imagePullSecrets: # Only needed if you're rate limited to Docker Hub
+  - regcred
+primary:
+  persistence:
+    enabled: true
+    storageClass: hpe-transform
+    annotations:
+      csi.hpe.com/importVolAsClone: virt-mm-db-1.vmdk
+    size: 64Gi
+    subPath: lib
+```
+
+The above parameters are quite self-explanatory, the `subPath` directive instructs the kubelet to mount the `lib` directory in the root of the volume. As a result, `/lib` in the volume will be mounted at `/bitnami/mariadb` and the `data` subdirectory which hold the MariaDB databases will be seen properly by the container binaries.
+
+Deploying the chart is straight forward.
+
+```markdown
+$ helm repo add bitnami https://charts.bitnami.com/bitnami
+$ helm install my-test-clone bitnami/mariadb -f values.yaml
+```
+
+Once the chart has been deployed and the database has come up, the “Hello World” test would of course be verifying the database.
+
+```markdown
+$ kubectl exec -it my-test-clone-mariadb-0 -- bash
+I have no name!@my-test-clone-mariadb-0:/$ mysql -ummattsson -ppassword < /bitnami/mariadb/scripts/test_employees_sha.sql
+INFO
+TESTING INSTALLATION
+computation_time
+00:00:49
+summary	result
+CRC	OK
+count	OK
+```
+
+If things dont’ come up properly, which is very likely it won’t on the first couple of iterations, read logs, error messages and course correct accordingly until the database come up cleanly. At this point, in a production scenario, test clients could be attached to further validate and solidify that moving the application to Kubernetes will actually work.
```

Comment:
This is a little confusing. Suggest rewriting to:
It's likely that things may not come up properly on the first couple of itterations. It needs to read logs, error messages, and course correct until the database comes up cleanly. Once it does, in a production scenario, test clients can be attached to further validate and solidify that moving the application to Kubernetes will actually work.